application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -65,26 +67,34 @@
         $response.setStatus(403)
       #end
     #end
-  #elseif("$!request.action" == "addtopic")
+  #elseif("$!request.action" == 'addtopic')
     #if("$!request.title.trim()" != '')
       #if("$!request.space" != '')
-        #set($newTopicName = $request.page)
         #set($topicTitle = "$!request.title.trim()")
-        #set($topicDoc = $xwiki.getDocument($newTopicName))
+        #set($topicSpace = "$!request.space")
+        #if($supportsNS)
+          #set($topicSpaceReference = $services.model.resolveSpace($topicSpace))
+          #set($forum = $topicSpaceReference.extractReference('SPACE').name)
+          #set($forumsRoot = $services.model.serialize($topicSpaceReference.parent))
+          #set($newDocReference = $services.model.createDocumentReference($doc.wiki, [$forumsRoot, $forum, $topicTitle], 'WebHome'))
```

Comment:
Also, remember that we`ve talked about the case when you are in a NS XWiki version but the forums data was not migrated so you also need to check if the current forum was migrated (i.e. if the forum's parent space is called "Forums").
If you don`t do this, when a user upgrade, but has not migrated his data yet, new topics will be created outside his forum.